packages/bootstrap-vue-next/src/components/BPopover.vue

```diff
@@ -1,198 +1,368 @@
-      >,
-      default: undefined,
-    },
-    title: {type: String},
-    delay: {type: [Number, Object] as PropType<number | BPopoverDelayObject>, default: 0},
-    triggers: {type: String as PropType<Popover.Options['trigger']>, default: 'click'},
-    show: {type: [Boolean, String] as PropType<Booleanish>, default: false},
-    variant: {type: String as PropType<ColorVariant>, default: undefined},
-    html: {type: [Boolean, String] as PropType<Booleanish>, default: true},
-    sanitize: {type: [Boolean, String] as PropType<Booleanish>, default: false},
-    offset: {type: String as PropType<Popover.Options['offset']>, default: '0'},
+interface BPopoverProps {
+  modelValue?: Booleanish
+  container?: string | ComponentPublicInstance<HTMLElement> | HTMLElement | null
+  target?:
+    | (() => HTMLElement | VNode)
+    | string
+    | ComponentPublicInstance<HTMLElement>
+    | HTMLElement
+    | null
+  reference?:
+    | (() => HTMLElement | VNode)
+    | string
+    | ComponentPublicInstance<HTMLElement>
+    | HTMLElement
+    | null
+  content?: string
+  id?: string
+  title?: string
+  delay?: number | {show: number; hide: number}
+  triggers?: string
+  variant?: ColorVariant
+  offset?: number
+  customClass?: string
+  placement?: Placement
+  strategy?: Strategy
+  floatingMiddleware?: Middleware[]
+  noFlip?: Booleanish
+  noShift?: Booleanish
+  noFade?: Booleanish
+  noAutoClose?: Booleanish
+  hide?: Booleanish
+  realtime?: Booleanish
+}
+
+const props = withDefaults(defineProps<BPopoverProps>(), {
+  modelValue: false,
+  container: null,
+  customClass: '',
+  placement: 'top',
+  strategy: 'absolute',
+  delay: () => ({show: 100, hide: 300}),
+  triggers: 'hover focus',
+  variant: undefined,
+  offset: 10,
+  noFlip: false,
+  noShift: false,
+  noFade: false,
+  hide: true,
+  realtime: false,
+  reference: null,
+  target: null,
+})
+
+interface BPopoverEmits {
+  (e: 'show'): void
+  (e: 'shown'): void
+  (e: 'hide'): void
+  (e: 'hidden'): void
+  (e: 'update:modelValue', value: boolean): void
+}
+
+const emit = defineEmits<BPopoverEmits>()
+
+// const id = useId('b-popover', props.id)
+
+// const showState = useVModel(props, 'modelValue', emit, {passive: true})
+const showState = ref(props.modelValue)
+watch(showState, (value) => {
+  emit('update:modelValue', !!value)
+})
+watch(
+  () => props.modelValue,
+  (value) => {
+    if (value === showState.value) return
+    value ? show() : dohide()
+  }
+)
+const noShiftBoolean = useBooleanish(toRef(props, 'noShift'))
+const noFlipBoolean = useBooleanish(toRef(props, 'noFlip'))
+const noFadeBoolean = useBooleanish(toRef(props, 'noFade'))
+const noAutoCloseBoolean = useBooleanish(toRef(props, 'noAutoClose'))
+const hideBoolean = useBooleanish(toRef(props, 'hide'))
+const realtimeBoolean = useBooleanish(toRef(props, 'realtime'))
+const hidden = ref(false)
+
+const elementRef = ref<HTMLElement>()
```

Comment:
```suggestion
const elementRef = ref<HTMLElement | null>(null)
```
This should apply to the rest.
Undefined and null work the same in these situations, but null is more explicit